packages/core/src/Provider/Provider.ts

```diff
@@ -3,6 +3,10 @@ export type HealthCheckResponse = {
 };
 
 export interface Provider {
+  start?(): Promise<void>;
```

Comment:
Looking at this again, I'm actually not liking core Provider having `{start, close}`. Can we move those to cardano-services? I suggest `DbSyncProvider` or a new interface that extends core Provider. What do you think @rhyslbw?

Reply:
Makes perfect sense: `type RunnableProvider = Provider & RunnableModule`?

Reply:
I like the suggestion of `RunnableProvider`. So in the end we agree on: 
1. Leave only `healthCheck` method within `Provider` and get rid of [{start, close}](https://github.com/input-output-hk/cardano-js-sdk/blob/2c61c6eb2a7a7de81ce37c5d524e526bc17767de/packages/core/src/Provider/Provider.ts#L6-L13)?
2. Rename `RunnableModule`'s [start](https://github.com/input-output-hk/cardano-js-sdk/blob/71a939b874296d86183d89fce7877c565630e921/packages/cardano-services/src/RunnableModule.ts#L28-L32) to `run` and all inners methods (before/after run)?
3. Create `type RunnableProvider = Provider & Pick<RunnableModule, 'run' | 'shutdown'>` within `cardano-services` package
4. These two changes should be a separate commit as part of this PR?

Reply:
Yes to all questions, unless @rhyslbw objects

Reply:
Just make sure to mark the method name change as breaking

Reply:
I think `runAfter` setting running state is confusing. Perhaps not such a great refactor idea :thinking:

Reply:
Yeah, since we have `'starting' | 'running' | 'stopping'` states, I guess we shouldn't ignore `start` as a runnable module lifecycle method?

Reply:
It turned out there is another issue with the refactoring - as discussed we remove `{start/stop}` methods from `core`'s [Provider](https://github.com/input-output-hk/cardano-js-sdk/blob/a2bfcc62c25e71d78d07b961267d7ce9679b6cf4/packages/core/src/Provider/Provider.ts#L5) interface and create `type RunnableProvider<T extends Provider> = T & Pick<RunnableModule, 'start' | 'shutdown'>` inside `cardano-services` and start using it. But we have another packages such as [rabbitmqTxSubmitProvider](https://github.com/input-output-hk/cardano-js-sdk/blob/3c0f6048c5cfa04654f0a5463dfccefd24c9054e/packages/rabbitmq/src/rabbitmqTxSubmitProvider.ts#L11) implementing [TxSubmitProvider](https://github.com/input-output-hk/cardano-js-sdk/blob/e155ed4efcd1338a54099d1a9034ccbeddeef1cc/packages/core/src/Provider/TxSubmitProvider/types.ts#L5) extending Provider. So the issue is we want to align all providers with `RunnableProvider` (located within `cardano-services`) where some of them depends only on the `core` package.
I'm happy to discuss the approach here in a quick huddle.